src/parser/Parser.ts

```diff
@@ -2418,17 +2466,70 @@ export class Parser {
         }
     }
 
+    private getBscTypeFromAssignment(assignment: AssignmentStatement, functionExpression: FunctionExpression): BscType {
+        try {
+            //function
+            if (isFunctionExpression(assignment.value)) {
+                let functionType = new FunctionType(assignment.value.returnType);
+                functionType.isSub = assignment.value.functionType.text === 'sub';
+                if (functionType.isSub) {
+                    functionType.returnType = new VoidType();
+                }
+
+                functionType.setName(assignment.name.text);
+                for (let param of assignment.value.parameters) {
+                    let isRequired = !param.defaultValue;
+                    //TODO compute optional parameters
+                    functionType.addParameter(param.name.text, param.type, isRequired);
+                }
+                return functionType;
+
+                //literal
+            } else if (isLiteralExpression(assignment.value)) {
+                return assignment.value.type;
+
+                //function call
+            } else if (isCallExpression(assignment.value)) {
+                let calleeName = ((assignment.value.callee as any).name.text as string).toLowerCase();
+                if (calleeName) {
+                    //make a lazy type which will not compute its type until the file is done parsing
+                    return new LazyType(() => {
```

Comment:
Ew I don't know about that...
Either the name can be resolved now, or it's a scope callable (which includes other top-level functions), and that doesn't require nearly as much magic closure work.

Reply:
This has been resolved in recent commits. @markwpearce created a `SymbolTable` that tracks this stuff at various levels of the program (Scope, Parser, FunctionExpression, NamespaceStatement)